src/elements/toniq-pagination/toniq-pagination.element.ts

```diff
@@ -0,0 +1,220 @@
+    },
+    initCallback: ({dispatch, events}) => {
+        dispatch(new events.created(undefined));
+    },
+    renderCallback: ({props, events, setProps, dispatch}) => {
+        /**
+         * Set minimum page size to 5. Lesser than 5 will show a [1 ...] or [1 ... ... 10] scenario
+         * which makes no sense.
+         */
+        const MINIMUM_PAGE_SIZE = 5;
+
+        setProps({
+            currentPage: clamp(props.currentPage, 1, props.pageCount),
+            pageSize: clamp(props.pageSize, MINIMUM_PAGE_SIZE, props.pageCount),
+        });
+
+        const getRange = (start: number, end: number) => {
+            const length = end - start + 1;
+            return Array.from({length}, (_, i) => start + i);
+        };
+
+        const pagination = (currentPage: number, pageCount: number, pageSize: number) => {
+            let delta: number;
+            currentPage = clamp(currentPage, 1, pageCount);
+            pageSize = clamp(pageSize, MINIMUM_PAGE_SIZE, pageCount);
+            const centerPageSize = pageSize - 5;
+            const boundaryPageSize = pageSize - 3;
+
+            if (pageCount <= pageSize) {
+                delta = pageSize;
+            } else {
+                delta =
+                    currentPage < boundaryPageSize || currentPage > pageCount - boundaryPageSize
+                        ? boundaryPageSize
+                        : centerPageSize;
+            }
+
+            const range = {
+                start: Math.round(currentPage - delta / 2),
+                end: Math.round(currentPage + delta / 2),
+            };
+
+            if (range.start - 1 === 1 || range.end + 1 === pageCount) {
+                range.start += 1;
+                range.end += 1;
+            }
+
+            let pages: (string | number)[] =
+                currentPage > delta
+                    ? getRange(
+                          Math.min(range.start, pageCount - delta),
+                          Math.min(range.end, pageCount),
+                      )
+                    : getRange(1, Math.min(pageCount, delta + 1));
+
+            if (currentPage > pageCount - boundaryPageSize) {
+                pages = getRange(pageCount - delta, pageCount);
+            }
+
+            const withDots = (value: number, pair: (string | number)[]) =>
+                pages.length + 1 !== pageCount ? pair : [value];
+            if (pages[0] !== 1) {
+                pages = withDots(1, [
+                    1,
+                    '...',
+                ]).concat(pages);
+            }
+            const lastPage = pages[pages.length - 1];
+            if (lastPage && lastPage < pageCount) {
+                pages = pages.concat(
+                    withDots(pageCount, [
+                        '...',
+                        pageCount,
+                    ]),
+                );
+            }
+
+            return pages;
+        };
+
+        return html`
+            <${ToniqButton}
```

Comment:
The pagination arrows should hide when there's only one page:

![pagination-arrows](https://user-images.githubusercontent.com/1205860/176802382-dc7abbe0-755a-4f99-8932-569b09cb1474.png)

Reply:
I think it would be better if we just hide the entire pagination? I think it would look awkward with just a page on the bottom with no controls see
![Screen Shot 2022-07-01 at 12 31 30 PM](https://user-images.githubusercontent.com/99699764/176823832-d01d207d-bd5e-4dd1-990f-1539fbbbf6ac.png)

Compare to no pagination
![Screen Shot 2022-07-01 at 12 32 58 PM](https://user-images.githubusercontent.com/99699764/176823908-b11dcdd0-1a58-4bc7-b451-9aca2cf4bf51.png)

Reply:
Agreed! Do you think this component should handle that logic or the parent should handle that logic? I could see both working just as fine.

Reply:
As a fail-safe the component should handle this logic as well